include/fiction/algorithms/simulation/sidb/critical_temperature.hpp

```diff
@@ -0,0 +1,378 @@
+            }
+        }
+
+        return true;
+    }
+
+    bool non_gate_based_simulation()
+    {
+
+        exgs_stats<Lyt> stats_exhaustive{};
+        exhaustive_ground_state_simulation(layout, parameter.sidb_sim_params, &stats_exhaustive);
+
+        // The number of physically valid charge configurations is stored.
+        temperature_stats.num_valid_lyt = stats_exhaustive.valid_lyts.size();
+
+        const auto distribution = energy_distribution(stats_exhaustive.valid_lyts);
+
+        // if there is more than one metastable state
+        if (temperature_stats.num_valid_lyt > 1)
+        {
+            const auto ground_state_energy        = distribution.cbegin()->first;
+            const auto first_excited_state_energy = std::next(distribution.cbegin())->first;
+
+            // The energy difference between the first excited and the ground state in meV.
+            temperature_stats.energy_between_ground_state_and_first_erroneous =
+                (first_excited_state_energy - ground_state_energy) * 1000;
+        }
+
+        std::vector<double> temp_values{};
+        temp_values.reserve(parameter.max_temperature * 100);
+
+        for (uint64_t i = 1; i <= parameter.max_temperature * 100; i++)
+        {
+            temp_values.push_back(static_cast<double>(i) / 100.0);
+        }
+
+        // This function determines the critical temperature (CT) for a given confidence level.
+        for (const auto& temp : temp_values)
+        {
+            // if the occupation probability of excited states exceeds the given threshold...
+            if (occupation_probability_non_gate_based(distribution, temp) > (1 - parameter.confidence_level))
+            {
+                // ...the current temperature is stored as the critical temperature
+                temperature_stats.critical_temperature = temp;
+
+                break;
+            }
+
+            if (std::abs(temp - static_cast<double>(parameter.max_temperature)) < 0.001)
+            {
+                // maximal temperature is stored as the critical temperature
+                temperature_stats.critical_temperature = static_cast<double>(parameter.max_temperature);
+            }
+        }
+
+        return true;
+    }
+
+  private:
+    Lyt&                             layout{};
+    critical_temperature_params&     parameter;
+    critical_temperature_stats<Lyt>& temperature_stats;
+};
+
+/**
+ * Computes the critical temperature of a given SiDB layout. If a gate is simulated, the temperature that results in a
+ * population of erroneous excited states with a probability greater than 1 − η, where η is the confidence level for the
+ * presence of a working gate, is called the *Critical Temperature (CT)* of the gate. In the case of an arbitrary layout
+ * (`gate_based_simulation` is set to false), temperatures above CT lead to a population of ground states smaller than
+ * η.
```

Comment:
Very nitpicky comment: maybe we can restructure the text a bit to guide the reader along the following steps:

1. this algorithm performs temperature-aware SiDB simulation
2. it comes in two flavors: gate-based and non-gate based; can be specified in the parameters (default = ...)
3. this is how gate-based works
4. this is how non-gate-based works

Also, maybe you could use MathJax for the symbols? That means you can use LaTeX syntax in the documentation by surrounding it with `\f$`, e.g., `\f$ 1 - \eta \f$`. This way, it is rendered as such in our online documentation.